test_rmw_implementation/test/test_subscription.cpp

```diff
@@ -377,3 +377,92 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_sequence) {
+  rmw_ret_t ret;
+  size_t count = 1u;
+  size_t taken;
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  rmw_message_sequence_t sequence = rmw_get_zero_initialized_message_sequence();
+  ret = rmw_message_sequence_init(&sequence, count, &allocator);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  rmw_message_info_sequence_t info_sequence = rmw_get_zero_initialized_message_info_sequence();
+  ret = rmw_message_info_sequence_init(&info_sequence, count, &allocator);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  rmw_subscription_allocation_t * null_allocation{nullptr}; // still valid allocator
+
+  ret = rmw_take_sequence(sub, count, &sequence, &info_sequence, &taken, null_allocation);
```

Comment:
I'm not familiarized with this API, but is the `taken` value updated on success? if that's the case you can add an extra check for that

Reply:
+1 to this.

Reply:
@hidmic but for `taken` to take a value != `0u`, I have to make sure the `subscriber` has at least one message right/

Reply:
Indeed, but you can set it to a nonzero value right before the call.

Reply:
Changed on 4a8441f